app/reflexes/volunteers_table_reflex.rb

```diff
@@ -1,11 +1,44 @@
 # frozen_string_literal: true
 
 class VolunteersTableReflex < ApplicationReflex
-  include SortHelper
+  include TableHelper
 
   def sort
-    sort_records(records: User.where(role: :volunteer),
-      partial: "volunteers_table",
-      sort_columns: %w[first_name last_name status])
+    sort_records(records: records, columns: %w[first_name last_name status])
+  end
+
+  def filter
+    filter_records(records: records,
+      columns: {"first_name" => :fuzzy,
+                "last_name" => :fuzzy,
+                "status" => :exact})
+  end
+
+  private
+
+  def records
+    return @records if defined?(@records)
+
+    @records = User.where(role: :volunteer)
```

Comment:
Same here. Can this be done in one line with `||=`?

Reply:
This is a common  pattern which is used to solve the `nil` problem

```
def records
  @records ||= ActiveRecord.query
end
```

If the `ActiveRecord.query` returns nil, in that case when next time the `records` method is called, the query will be evaluated again. 

https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/

Reply:
Cool! I hadn't thought of that.

In this case though, and the above, the result will never be `nil`. If `@records` were `nil`, then `@records.each` on the page would error, because it has to be a collection. Even in a db with no `User` records at all, `User.where(role: :volunteer)` will return an empty AR collection.

And
```ruby
@sort_direction ||= %w[asc desc].include?(element.dataset.direction) ? element.dataset.direction : "asc"
```
can only ever be `"asc"` or `"desc"`.